tendermint/src/lite/verifier.rs

```diff
@@ -55,84 +95,34 @@ where
 }
 
 /// Verify the commit is valid from the given validators for the header.
-pub fn verify<H, V, C>(header: H, commit: C, validators: V) -> Result<(), Error>
+pub fn verify<SH, V>(signed_header: &SH, validators: &V) -> Result<(), Error>
 where
-    H: Header,
+    SH: SignedHeader,
     V: ValidatorSet,
-    C: Commit,
 {
-    if let Err(e) = validate_vals_and_commit(header, &commit, &validators) {
+    let header = signed_header.header();
+    let commit = signed_header.commit();
+    if let Err(e) = validate_vals_and_commit(header, commit, validators) {
         return Err(e);
     }
 
     // ensure that +2/3 validators signed correctly
-    verify_commit_full(&validators, commit)
-}
-
-/// Verify the commit is trusted according to the last validators and is valid
-/// from the current validators for the header.
-pub fn verify_trusting<H, V, C>(
-    header: H,
-    commit: C,
-    last_validators: V,
-    validators: V,
-) -> Result<(), Error>
-where
-    H: Header,
-    V: ValidatorSetLookup,
-    C: Commit,
-{
-    // NOTE it might be more prudent to do the cheap validations first
-    // before we even call verify_commit_trusting, but not doing that
-    // makes the code cleaner and allows us to just call verify directly.
-
-    // ensure that +1/3 of last trusted validators signed correctly
-    if let Err(e) = verify_commit_trusting(&last_validators, &commit) {
-        return Err(e);
-    }
-
-    // perform same verification as in sequential case
-    verify(header, commit, validators)
+    verify_commit_full(validators, commit)
 }
 
 /// Verify that +2/3 of the correct validator set signed this commit.
 /// NOTE: these validators are expected to be the correct validators for the commit.
-fn verify_commit_full<V, C>(vals: &V, commit: C) -> Result<(), Error>
+fn verify_commit_full<V, C>(vals: &V, commit: &C) -> Result<(), Error>
 where
     V: ValidatorSet,
     C: Commit,
 {
     let total_power = vals.total_power();
-    let mut signed_power: u64 = 0;
-
-    let vals_vec = vals.into_vec();
-    let commit_vec = commit.into_vec();
-
-    if vals_vec.len() != commit_vec.len() {
+    if vals.len() != commit.votes_len() {
```

Comment:
Should we make this part of `validate_vals_and_commit` instead ?

Also, should we make that function a method on `Commit` and then we can drop the need for a `votes_len` and hide the validity checks from this package, as per the spec comment `Additional checks should be done in the implementation to ensure header is well formed`

Reply:
Good point! tracked in #96